dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -237,6 +244,63 @@ public ChainedContext getContext(final HttpServletRequest request, final HttpSer
 		return getWebContext(getBasicContext(), request, response);
 	}
 	
+  /**
+   * This will return a velocity context for workflow actionlet.
+   * It will mock a Request and Response and then use
+   */
+  public Context getWorkflowContext(final WorkflowProcessor processor) {
+    
+    final Contentlet contentlet = processor.getContentlet();
+    final ContentType contentType = contentlet.getContentType();
+    final Host host =  Try
+        .of(() -> Host.SYSTEM_HOST.equals(contentlet.getHost()) || null == contentlet.getHost() 
+        ? APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false)
+            : APILocator.getHostAPI().find(contentlet.getHost(), APILocator.systemUser(), false))
+        .getOrElse(APILocator.systemHost());
+    
+    final HttpServletRequest requestProxy= (HttpServletRequestThreadLocal.INSTANCE.getRequest() != null) ? HttpServletRequestThreadLocal.INSTANCE.getRequest()
```

Comment:
this would be great to create a Builder
```
Context context = new ContextBuilder().put("host",host).put("contentType", contentType).put("host_id", host.getIdentifier())
....put("contentMap", new ContentMap(contentlet, processor.getUser(),PageMode.PREVIEW_MODE,host,context)
.build();
```